src/serde_macros.rs

```diff
@@ -0,0 +1,42 @@
+
+#[cfg(feature="serde")]
+macro_rules! serde_impl(
```

Comment:
Since we noticed that we might be able to use `1.24.0` we can probably refactor out the `$len` parameter.

Reply:
On further reflection I think we probably don't want to go too far past `1.19` which is the currently-supported mrustc version. Though `1.20` is not too bad, it only requires one additional layer of bootstrapping to avoid "trusting trust" attacks; this would give us associated constants at least.